libnmstate/nm/ethtool.py

```diff
@@ -0,0 +1,58 @@
+#
+# Copyright (c) 2018-2021 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from libnmstate.error import NmstateDependencyError
+from .common import NM
+from .common import GLib
+
+
+def create_ethtool_setting(iface_ethtool, base_con_profile):
+    nm_setting = None
+    if not hasattr(NM, "ETHTOOL_OPTNAME_PAUSE_AUTONEG"):
+        raise NmstateDependencyError(
+            "Ethtool PAUSE is only supported by NetworkManager 1.31+"
+        )
+    if base_con_profile:
+        nm_setting = base_con_profile.get_setting_by_name(
+            NM.SETTING_ETHTOOL_SETTING_NAME
+        )
+        if nm_setting:
+            nm_setting = nm_setting.duplicate()
+
+    if not nm_setting:
+        nm_setting = NM.SettingEthtool.new()
+
+    if iface_ethtool.pause:
+        if iface_ethtool.pause.autoneg is not None:
+            nm_setting.option_set(
+                NM.ETHTOOL_OPTNAME_PAUSE_AUTONEG,
```

Comment:
I think this is not correct. The linter complains because NetworkManager does not have that attribute.